tools/mason/MasonBuild.chpl

```diff
@@ -64,54 +68,63 @@ private proc checkChplVersion(lockFile : Toml) {
   }
 }
 
-proc buildProgram(release: bool, show: bool, compopts: [?d] string,
+proc buildProgram(release: bool, show: bool, force: bool, compopts: [?d] string,
                   tomlName="Mason.toml", lockName="Mason.lock") {
 
   try! {
 
     const cwd = getEnv("PWD");
     const projectHome = getProjectHome(cwd, tomlName);
-
-    if isFile(projectHome + "/" + lockName) {
-
-      // --fast
-      var binLoc = 'debug';
-      if release then
-        binLoc = 'release';
-
-      // Make Binary Directory
-      makeTargetFiles(binLoc, projectHome);
-
-      // Install dependencies into $MASON_HOME/src
-      var toParse = open(projectHome + "/" + lockName, iomode.r);
-      var lockFile = new Owned(parseToml(toParse));
-      checkChplVersion(lockFile);
-
-      if isDir(MASON_HOME) == false {
-        mkdir(MASON_HOME, parents=true);
-      }
-
-      var sourceList = genSourceList(lockFile);
-      getSrcCode(sourceList, show);
-
-      // Checks if dependencies exist and retrieves them for compilation
-      if lockFile.pathExists('root.compopts') {
-        const cmpFlags = lockFile["root"]["compopts"].s;
-        compopts.push_back(cmpFlags);
-      }
-
-      // Compile Program
-      if compileSrc(lockFile, binLoc, show, release, compopts, projectHome) {
-        writeln("Build Successful\n");
+    const toParse = open(projectHome + "/" + lockName, iomode.r);
+    var lockFile = new Owned(parseToml(toParse));
+    const projectName = lockFile["root"]["name"].s;
+    
+    // --fast
+    var binLoc = 'debug';
+    if release then
+      binLoc = 'release';
+
+
+    // build on last modification
+    if projectModified(projectHome, projectName, binLoc) || force {
+
+      if isFile(projectHome + "/" + lockName) {
+
+        // Make build files and check chapel version
+        makeTargetFiles(binLoc, projectHome);
+        checkChplVersion(lockFile);
+
+        if isDir(MASON_HOME) == false {
+          mkdir(MASON_HOME, parents=true);
+        }
+
+        // generate list of dependencies and get src code
+        var sourceList = genSourceList(lockFile);
+        getSrcCode(sourceList, show);
+
+        // Checks for compilation options are present in Mason.toml
+        if lockFile.pathExists('root.compopts') {
+          const cmpFlags = lockFile["root"]["compopts"].s;
+          compopts.push_back(cmpFlags);
+        }
+
+        // Compile Program
+        if compileSrc(lockFile, binLoc, show, release, compopts, projectHome) {
+          writeln("Build Successful\n");
+        }
+        else {
+          writeln("Build Failed");
+          exit(1);
+        }
+        // Close memory
+        toParse.close();
       }
-      else {
-        writeln("Build Failed");
-        exit(1);
-      }
-      // Close memory
-      toParse.close();
+      else writeln("Cannot build: no Mason.lock found");
+      exit(1);
+    }
+    else {
+      writeln("Build stopped: No changes to project");
```

Comment:
The text "Build stopped: ..." comes off as an error message, like "progress has been halted". 

It may be less alarming to users to change this to something like "Build skipped: ..." or "Skipping build: ...".